app/javascript/miq-redux/store.ts

```diff
@@ -0,0 +1,19 @@
+import { createStore, applyMiddleware, compose, Store} from 'redux';
+import {rootReducer} from './reducer';
+
+import {AppState} from '../packs/miq-redux';
+
+const composeEnhancers = window['.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] || compose;
```

Comment:
Since the `__REDUX_DEVTOOLS_EXTENSION_COMPOSE__` global is well-known in advance, I'd prefer the following form:

```
window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
```

and tell TypeScript to accept that global. I'd use the `[]` access mainly for dynamic properties.

Since [Redux `compose`](https://github.com/reactjs/redux/blob/master/src/compose.js) returns a function, the TypeScript-annotated property access should be `(<function>window)`.

Looking at [TypeScript declaration docs](https://basarat.gitbooks.io/typescript/docs/types/ambient/d.ts.html), the proper solution should be to declare `__REDUX_DEVTOOLS_EXTENSION_COMPOSE__` as global:

```
export namespace ManageIQ {
  // export "redux" here
}
```